app/models/match_conversation.rb

```diff
@@ -2,38 +2,66 @@
 #
 # Table name: match_conversations
 #
-#  id         :integer          not null, primary key
-#  topic_id   :integer
-#  user_A_id  :integer
-#  user_B_id  :integer
-#  created_at :datetime         not null
-#  updated_at :datetime         not null
+#  id            :integer          not null, primary key
+#  topic_id      :integer
+#  user_a_id     :integer
+#  user_b_id     :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#  channel_id    :string
+#  last_logout_a :datetime
+#  last_logout_b :datetime
 #
 
 class MatchConversation < ActiveRecord::Base
-  belongs_to :user_A, :class_name => 'User'
-  belongs_to :user_B, :class_name => 'User'
+  belongs_to :user_a, :class_name => 'User'
+  belongs_to :user_b, :class_name => 'User'
   belongs_to :topic
   has_many   :messages
 
-  validate   :match_uniqueness
+  validate   :match_uniqueness, on: :create
+  validates  :channel_id, uniqueness: true
 
-  after_save :notify
+  before_validation :set_channel_id, on: [:create, :update]
+  after_save        :notify
 
   def match_uniqueness
     matches_topic = MatchConversation.where(topic_id: topic_id)
-    if matches_topic.where(user_A_id: [user_A_id, user_B_id]).where(user_B_id: [user_A_id, user_B_id]).any?
+    if matches_topic.where(user_a_id: [user_a_id, user_b_id]).where(user_b_id: [user_a_id, user_b_id]).any?
       errors.add(:already_matched, "Already Matched")
     end
   end
 
-  def other_party
-    return current_user.id == user_B.id ? user_A : user_B
+  def other_party(current_user)
+    return current_user.id == user_a.id ? user_b : user_a
+  end
+
+  def unreaded(user)
+    last_logout = user_a_id == user.id ? last_logout_a : last_logout_b
+    if last_logout.nil?
+      messages.all
+    else
+      messages.after(user_a_id == user.id ? last_logout_a : last_logout_b)
```

Comment:
Couldn't you change 

`user_a_id == user.id ? last_logout_a : last_logout_b`

for

`last_logout` ??